code/src/hooks/use-filters-with-apply/useFiltersWithApply.test.ts

```diff
@@ -0,0 +1,160 @@
+
+import parseFiltersFromParams from "@/hooks/use-filters-with-apply/parse-filters-from-params/parseFiltersFromParams";
+import serializeToQueryString from "@/hooks/use-filters-with-apply/serialize-to-query-string/serializeToQueryString";
+import useFiltersWithApply from "@/hooks/use-filters-with-apply/useFiltersWithApply";
+
+jest.mock("react-router-dom", () => ({
+  useSearchParams: jest.fn()
+}));
+
+jest.mock(
+  "@/hooks/use-filters-with-apply/parse-filters-from-params/parseFiltersFromParams"
+);
+jest.mock(
+  "@/hooks/use-filters-with-apply/serialize-to-query-string/serializeToQueryString"
+);
+
+const mockSetSearchParams = jest.fn();
+
+describe("useFiltersWithApply", () => {
+  let defaultFilters: Record<string, unknown>;
+
+  beforeEach(() => {
+    defaultFilters = { filter1: "default1", filter2: "default2" };
+
+    (useSearchParams as jest.Mock).mockReturnValue([
+      new URLSearchParams(),
+      mockSetSearchParams
+    ]);
+
+    (parseFiltersFromParams as jest.Mock).mockReturnValue({
+      defaultActiveFilters: new Set(),
+      defaultFiltersFromParams: defaultFilters
+    });
+
+    (serializeToQueryString as jest.Mock).mockImplementation((value: object) =>
+      value.toString()
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("initializes with default filters correctly", () => {
+    const { result } = renderHook(() => useFiltersWithApply(defaultFilters));
+
+    expect(result.current.filters).toEqual(defaultFilters);
+    expect(result.current.appliedFilters).toEqual({});
+    expect(result.current.activeFiltersCount).toBe(0);
+  });
+
+  test("updates a filter by key correctly", () => {
+    const { result } = renderHook(() => useFiltersWithApply(defaultFilters));
+
+    act(() => {
+      result.current.actions.updateFilterByKey("filter1", "updated1");
+    });
+
+    expect(result.current.filters.filter1).toBe("updated1");
+    expect(result.current.activeFiltersCount).toBe(1);
+  });
+
+  test("resets a filter by key correctly", () => {
+    const { result } = renderHook(() => useFiltersWithApply(defaultFilters));
+
+    act(() => {
+      result.current.actions.updateFilterByKey("filter1", "updated1");
+      result.current.actions.resetFilterByKey("filter1");
+    });
+
+    expect(result.current.filters.filter1).toBe("default1");
+    expect(result.current.activeFiltersCount).toBe(0);
+  });
+
+  test("resets all filters correctly", () => {
+    const { result } = renderHook(() => useFiltersWithApply(defaultFilters));
+
+    act(() => {
+      result.current.actions.updateFilterByKey("filter1", "updated1");
+      result.current.actions.updateFilterByKey("filter2", "updated2");
+      result.current.actions.resetFilters();
+    });
+
+    expect(result.current.filters).toEqual(defaultFilters);
+    expect(result.current.activeFiltersCount).toBe(0);
+  });
+
+  test("applies filters correctly", () => {
+    const { result } = renderHook(() => useFiltersWithApply(defaultFilters));
```

Comment:
check it uses it each test. Can improve it

Reply:
Tried ny best to improve it. Please rereview

Reply:
> Tried ny best to improve it. Please rereview

I just saw it's repeated also in `renderAndMock`
We should move on and not concentrate for that I guess